docs/deploying.md

```diff
@@ -1,35 +1,419 @@
+			<td>TAG</td>
+			<td>IMAGE ID</td>
+			<td>CREATED</td>
+			<td>VIRTUAL SIZE</td>
+		</tr>
+		<tr>
+			<td>...</td>
+			<td>...</td>
+			<td>...</td>
+			<td>...</td>
+			<td>...</td>
+		</tr>           
+		<tr>
+			<td>localhost:5000/hello-mine</td>
+			<td>latest</td>
+			<td>e45a5af57b00</td>
+			<td>3 months ago</td>
+			<td>910 B</td>
+		</tr>
+	</table>
+
+5. Push this new image to your local registry.
+
+		$ docker push localhost:5000/hello-mine:latest
+			
+6. Remove all the unused images from your local environment:
+
+		$ docker rmi -f $(docker images -q -a )
+
+	This command is for illustrative purposes; removing the image forces any `run`
+	to pull from a registry rather than a local cache. If you run `docker images`
+	after this you should not see any instance of `hello-world` or `hello-mine` in
+	your images list.
+	<table class="code">
+      <tr>
+        <td>REPOSITORY</td>
+        <td>TAG</td>
+        <td>IMAGE ID</td>
+        <td>CREATED</td>
+        <td>VIRTUAL SIZE</td>
+      </tr>
+      <tr>
+        <td>registry</td>
+        <td>2.0</td>
+        <td>bbf0b6ffe923</td>
+        <td>3 days ago</td>
+        <td>545.1 MB</td>
+      </tr>        
+      <tr>
+        <td>golang</td>
+        <td>1.4</td>
+        <td>121a93c90463</td>
+        <td>5 days ago</td>
+        <td>514.9 MB</td>
+      </tr>      
+    </table>
+	
+7. Try running `hello-mine`.
+
+		$ docker run hello-mine
+		Unable to find image 'hello-mine:latest' locally
+		Pulling repository hello-mine
+		FATA[0001] Error: image library/hello-mine:latest not found 
+		
+	The `run` command fails because your new image doesn't exist in the Docker public
+	registry.
+
+8. Now, try running the image but specifying the image's registry:
+
+		$ docker run localhost:5000/hello-mine
+
+	If you run `docker images` after this you'll fine a `hello-mine` instance. 
+		
+### The official image is not production ready
+
+Docker's official image is for simple tests or debugging. Its configuration is
+unsuitable for production. For example, any client with access to the server's
+IP can push and pull images to it. The official image also uses a location in
+the container's filesystem to store images. These are just two ways in which the 
+official image is insufficient for production.
+
+## Considerations for deployment 
+
+When deploying a registry you should consider these factors:
+
+* are you serving images to the public or internally to your company only
+* should the service reside on a hosted or in-house server
+* do users should have full or controlled access
+* what kind of storage to use for your images
+
+You can configure your repository features to adjust for these factors. You do
+this by specifying options on the command line or, more typically, by writing a
+registry configuration file. The configuration file is in YAML format.
+
+Docker's official repository image it is preconfigured using the following
+configuration file:
+
+```yaml
```

Comment:
This yaml spacing may be problematic. I know the documentation is using tabs but we should really convert it all to two spaces to make it pastable into a configuration file. For example, this should look like this:

```yaml
version: 0.1
log:
  level: debug
  fields:
    service: registry
    environment: development
storage:
    cache:
        layerinfo: inmemory
    filesystem:
        rootdirectory: /tmp/registry-dev
http:
    addr: :5000
    secret: asecretforlocaldevelopment
    debug:
        addr: localhost:5001
redis:
  addr: localhost:6379
  pool:
    maxidle: 16
    maxactive: 64
    idletimeout: 300s
  dialtimeout: 10ms
  readtimeout: 10ms
  writetimeout: 10ms
notifications:
  endpoints:
      - name: local-8082
        url: http://localhost:5003/callback
        headers:
           Authorization: [Bearer <an example token>]
        timeout: 1s
        threshold: 10
        backoff: 1s
        disabled: true
      - name: local-8083
        url: http://localhost:8083/callback
        timeout: 1s
        threshold: 10
        backoff: 1s
        disabled: true
```

Check the raw source of this comment to get the markdown formatting.